UWP/UWPHelpers/StorageManager.cpp

```diff
@@ -0,0 +1,331 @@
+
+	FILE* file{};
+
+	auto pathResolved = Path(ResolvePathUWP(path));
+	HANDLE handle;
+	auto access = GENERIC_READ;
+	auto share = FILE_SHARE_READ;
+	auto creation = OPEN_EXISTING;
+	bool isWrite = isWriteMode(mode);
+	bool isAppend = isAppendMode(mode);
+
+	if (isWrite) {
+		access = GENERIC_WRITE;
+		share = FILE_SHARE_WRITE;
+		creation = isAppend ? OPEN_ALWAYS : CREATE_ALWAYS;
+	}
+	handle = CreateFile2FromAppW(pathResolved.ToWString().c_str(), access, share, creation, nullptr);
+
+	if (handle != INVALID_HANDLE_VALUE) {
+		int flags = _O_RDONLY;
+		if (isWrite) {
+			flags = _O_RDWR;
+		}
+		file = _fdopen(_open_osfhandle((intptr_t)handle, flags), mode);
+	}
+
+	return file;
+}
+
+#pragma endregion
+
+#pragma region FakeFolders
+// Parent and child full path
+std::string getSubRoot(std::string parent, std::string child) {
+	auto childCut = child;
+	childCut = ReplaceAll(childCut, (parent + "/"), "");
+	size_t len = childCut.find_first_of('/', 0);
+	auto subRoot = childCut.substr(0, len);
+
+	return parent + "/" + subRoot;
+}
+
+bool isChild(std::string parent, std::string child) {
+	return child.find(parent) != std::string::npos;
+}
+
+// Parent full path, child full path, child name only
+bool isParent(std::string parent, std::string child, std::string childName) {
+	parent.append("/" + childName);
+	return parent == child;
+}
+
+bool IsRootForAccessibleItems(Path path, std::list<std::string>& subRoot, bool breakOnFirstMatch = false) {
+	path = PathResolver(path);
+	auto FutureAccessItems = GetFutureAccessList();
+	for (auto& fItem : FutureAccessItems) {
+		if (isChild(path.ToString(), fItem)) {
+			if (breakOnFirstMatch) {
+				// Just checking, we don't need to loop for each item
+				return true;
+			}
+			auto sub = getSubRoot(path.ToString(), fItem);
+
+			// This check can be better, but that's how I can do it in C++
+			if (!endsWith(sub, ":")) {
+				bool alreadyAdded = false;
+				for each (auto sItem in subRoot) {
+					if (!strcmp(sItem.c_str(), sub.c_str())) {
+						alreadyAdded = true;
+						break;
+					}
+				}
+				if (!alreadyAdded) {
+					subRoot.push_back(sub);
+				}
+			}
+		}
+	}
+	return !subRoot.empty();
+}
+bool IsRootForAccessibleItems(std::string path)
+{
+	std::list<std::string> tmp;
+	return IsRootForAccessibleItems(Path(path), tmp, true);
+}
+
+bool GetFakeFolders(Path path, std::vector<File::FileInfo>* files, const char* filter, std::set<std::string> filters) {
+	bool state = false;
+	std::list<std::string> subRoot;
+	if (IsRootForAccessibleItems(path, subRoot)) {
+		if (!subRoot.empty()) {
+			for each (auto sItem in subRoot) {
+				auto folderPath = Path(sItem);
+				auto attributes = FILE_ATTRIBUTE_DIRECTORY;
+				File::FileInfo info;
+				info.name = folderPath.GetFilename();
+				info.fullName = folderPath;
+				info.exists = true;
+				info.size = 1;
+				info.isDirectory = true;
+				info.isWritable = (attributes & FILE_ATTRIBUTE_READONLY) == 0;
+				info.atime = 1000;
```

Comment:
0 might be a better "I don't know".

-[Unknown]

Reply:
Oh, I meant for the atime/mtime/ctime.  We might actually want isWritable to be true - this is only used by some games when they run `stat()`, but for that one `true` is probably safest if we don't know.

-[Unknown]